community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java

```diff
@@ -276,11 +280,16 @@ public CapableIndexReference index( int label, int... properties )
     {
         ktx.assertOpen();
 
-        SchemaIndexDescriptor indexDescriptor =
-                storeReadLayer.indexGetForSchema( new LabelSchemaDescriptor( label, properties ) );
+        if ( label == KeyReadOperations.NO_SUCH_LABEL
```

Comment:
how come that so deep on this level we need to check for label and property id validity?

Reply:
I don't know. I'm just preserving the existing behaviour.

Reply:
because current 3.4 does not have that and it does not look right

Reply:
The `SchemaDescriptorFactory` does not allow creating descriptors with obviously invalid ids and will throw an exception on the attempt. And there are tests that verify that this method returns `CapableIndexReference.NO_INDEX` in such cases. I have changed the code to catch the exception instead.